src/ProgressOnderwijsUtils.Analyzers/UseAsEnrolledSqlInExpressionAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace ProgressOnderwijsUtils.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseAsEnrolledSqlInExpressionAnalyzer : DiagnosticAnalyzer
+{
+    public static readonly DiagnosticDescriptor Rule = new(
+#pragma warning disable RS2008 // Enable analyzer release tracking - why would we care?
+        "POU1010",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+        "Use IEnumerable<>.AsEnrolledSqlInExpression()",
+        "Use the more efficient IEnumerable<>.AsEnrolledSqlInExpression() for in expressions on small sets of constant size.",
+        "Functional",
+        DiagnosticSeverity.Error,
+        true
+    );
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        => ImmutableArray.Create(Rule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
```

Comment:
Dit zet analyze op generated code aan, lijkt me - willen we dat?  Vast niet toch?